lib/api/objectPutPart.js

```diff
@@ -87,80 +60,158 @@ export default function objectPutPart(authInfo, request, streamingV4Params,
     if (!Number.isInteger(partNumber) || partNumber < 1) {
         return cb(errors.InvalidArgument);
     }
-    // If part size is greater than 5GB, reject it
-    if (Number.parseInt(size, 10) > 5368709120) {
-        return cb(errors.EntityTooLarge);
-    }
-    // Note: Parts are supposed to be at least 5MB except for last part.
-    // However, there is no way to know whether a part is the last part
-    // since keep taking parts until get a completion request.  But can
-    // expect parts of at least 5MB until last part.  Also, we check that
-    // part sizes are large enough when mutlipart upload completed.
-
-    // Note that keys in the query object retain their case, so
-    // request.query.uploadId must be called with that exact
-    // capitalization
-    const uploadId = request.query.uploadId;
-    const metadataValMPUparams = {
-        authInfo,
-        bucketName,
-        objectKey,
-        uploadId,
-        requestType: 'putPart or complete',
-        log,
-        splitter: constants.splitter,
-    };
-    // For validating the request at the destinationBucket level
-    // params are the same as validating at the MPU level
-    // but the requestType is the more general 'objectPut'
-    const metadataValParams = Object.assign({}, metadataValMPUparams);
-    metadataValParams.requestType = 'objectPut';
+    const bucketName = request.bucketName;
+    assert.strictEqual(typeof bucketName, 'string');
+    const canonicalID = authInfo.getCanonicalID();
+    assert.strictEqual(typeof canonicalID, 'string');
     log.trace('owner canonicalid to send to data', {
         canonicalID: authInfo.getCanonicalID,
     });
-    const objectKeyContext = {
-        bucketName,
-        owner: authInfo.getCanonicalID(),
-        namespace: request.namespace,
-    };
-
-    return services.metadataValidateAuthorization(
-        metadataValParams,
-        (err, bucket) => {
+    // Note that keys in the query object retain their case, so
+    // `request.query.uploadId` must be called with that exact capitalization.
+    const uploadId = request.query.uploadId;
+    const mpuBucketName = `${constants.mpuBucketPrefix}${bucketName}`;
+    const objectKey = request.objectKey;
+    // If bucket has no server-side encryption, `cipherBundle` remains `null`.
+    let cipherBundle = null;
+    let splitter = constants.splitter;
+    return async.waterfall([
+        // Get the destination bucket.
+        next => metadata.getBucket(bucketName, log, (err, bucket) => {
+            if (err && err.NoSuchBucket) {
+                return next(errors.NoSuchUpload);
```

Comment:
Thanks, good catch! I have updated this to return a `NoSuchBucket` error as is the expected behavior. A `NoSuchUpload` error in this API should only be returned when the MPU bucket is not found.